### PR TITLE
cleanup, removes facade & dead code

### PR DESCRIPTION
**Pull request by vicb**

### CODE REVIEW

File: modules/@angular/platform-webworker/src/web_workers/shared/serializer.ts [hunk 1]
```diff
@@ -54,26 +53,32 @@ export class Serializer {
     if (!isPresent(map)) {
       return null;
     }
-    if (isArray(map)) {
-      var obj: any[] = [];
-      (<any[]>map).forEach(val => obj.push(this.deserialize(val, type, data)));
-      return obj;
+
+    if (Array.isArray(map)) {
+      return (<any[]>map).map(val => this.deserialize(val, type, data));
```
Comment (IgorMinar) [inline modules/@angular/platform-webworker/src/web_workers/shared/serializer.ts]: do we still need to cast? isn't isArray a type guard?

File: modules/@angular/platform-webworker/src/web_workers/shared/serializer.ts [hunk 2]
```diff
@@ -54,26 +53,32 @@ export class Serializer {
...
+
+    if (type === RenderComponentType) {
       return this._deserializeRenderComponentType(map);
-    } else if (type === ViewEncapsulation) {
-      return VIEW_ENCAPSULATION_VALUES[map];
-    } else if (type === LocationType) {
+    }
+
+    if (type === ViewEncapsulation) {
+      return map as ViewEncapsulation;
```
Comment (IgorMinar) [inline modules/@angular/platform-webworker/src/web_workers/shared/serializer.ts]: this change doesn't seem to be equivalent. what's going on here?

**Review by IgorMinar — APPROVED** [review]

File: modules/@angular/platform-webworker/src/web_workers/shared/serializer.ts (hunk 1 above)
Comment (vicb) [inline modules/@angular/platform-webworker/src/web_workers/shared/serializer.ts]: might be needed for older TS

File: modules/@angular/platform-webworker/src/web_workers/shared/serializer.ts (hunk 2 above)
Comment (vicb) [inline modules/@angular/platform-webworker/src/web_workers/shared/serializer.ts]: resolved with Igor

**Comment by rkirov** [comment]
needs rebase

**Pull request merged**